src/test/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/integration/MergeOffenderTest.kt

```diff
@@ -45,23 +51,49 @@ class MergeOffenderTest : SqsIntegrationTestBase() {
     )
 
     awaitAtMost30Secs untilAsserted {
-      verify(telemetryClient).trackEvent(
-        MERGE_EVENT_NAME,
-        mapOf(
-          "NOMS-MERGE-FROM" to OLD_PRISON_NUMBER,
-          "NOMS-MERGE-TO" to NEW_PRISON_NUMBER,
-          "UPDATED-RECORDS" to "3",
-        ),
-        null,
-      )
+      verify(done).complete()
     }
 
+    verify(telemetryClient).trackEvent(
+      MERGE_EVENT_NAME,
+      mapOf(
+        "NOMS-MERGE-FROM" to OLD_PRISON_NUMBER,
+        "NOMS-MERGE-TO" to NEW_PRISON_NUMBER,
+        "UPDATED-RECORDS" to "3",
+      ),
+      null,
+    )
+
     assertThat(licenceRepository.findAllByPrisonNumber(OLD_PRISON_NUMBER)).hasSize(0)
     assertThat(licenceRepository.findAllByPrisonNumber(NEW_PRISON_NUMBER)).hasSize(4)
 
     assertThat(getNumberOfMessagesCurrentlyOnQueue()).isEqualTo(0)
   }
 
+  @Test
+  @Sql(
+    "classpath:test_data/reset.sql",
+    "classpath:test_data/merge-offender.sql",
+  )
+  fun checkNoEventWhenNoRecordsToUpdate() {
+    val someNonExistentPrisonNumber = "ZZ1234AA"
+    assertThat(licenceRepository.findAllByPrisonNumber(someNonExistentPrisonNumber)).hasSize(0)
+
+    publishDomainEventMessage(
+      PRISONER_MERGE_EVENT_TYPE,
+      AdditionalInformationMerge(removedNomsNumber = someNonExistentPrisonNumber, nomsNumber = NEW_PRISON_NUMBER),
+      "A prisoner has been merged from $someNonExistentPrisonNumber to $NEW_PRISON_NUMBER",
+    )
+
+    awaitAtMost30Secs untilAsserted {
+      verify(done).complete()
+    }
```

Comment:
ensure that message has been processed before continuing..

we can't wait on this: 
`assertThat(getNumberOfMessagesCurrentlyOnQueue()).isEqualTo(0)`

as this would be true before the message has been fully published to the queue